qa/rpc-tests/excessive.py

```diff
@@ -19,40 +15,79 @@
 import test_framework.script as script
 import pdb
 import sys
-if sys.version_info[0] < 3: raise "Use Python 3"
+if sys.version_info[0] < 3:
+    raise "Use Python 3"
 import logging
 logging.basicConfig(format='%(asctime)s.%(levelname)s: %(message)s', level=logging.INFO)
 
+
+def mostly_sync_mempools(rpc_connections, difference=50, wait=1, verbose=1):
+    """
+    Wait until everybody has the most of the same transactions in their memory
+    pools. There is no guarantee that mempools will ever sync due to the
+    filterInventoryKnown bloom filter.
+    """
+    iterations = 0
+    while True:
+        iterations += 1
+        pool = set(rpc_connections[0].getrawmempool())
+        num_match = 1
+        poolLen = [len(pool)]
+        for i in range(1, len(rpc_connections)):
+            tmp = set(rpc_connections[i].getrawmempool())
+            if tmp == pool:
+                num_match = num_match + 1
+            if iterations > 10 and len(tmp.symmetric_difference(pool)) < difference:
+                num_match = num_match + 1
```

Comment:
yes this is only local to excessive.py.  But we have a systemic problem which is that the mempools may NEVER sync if there are any tx in the network because the bloom filter that is used to detect whether a tx is needed can have false negatives.  We briefly slack chatted about this problem about a week ago.  Basically if you run any test that uses sync_all() with lots of tx in the mempool many times, you'll get random hangs rarely.  We need to fix this across the qa tests to make our travis runs more consistent.